### PR TITLE
.Net Core support and Cake build support

### PR DESCRIPTION
**Pull request by gpailler**
I introduced new csproj files compatible with VS2017 RC (SlackApi.Netcore.csproj...) because format project.json is abandoned.
Nuget package contains net45, netstandard1.3 and netstandard1.6 assemblies (netstandard1.3 version doesn't support detection of RoutingMessage in external assemblies but is compatible with more platforms like UWP. See https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/ for details)

To build using cake:
- Download and install latest LTS .NET Core SDK version 1.0.0-preview4-004233
https://github.com/dotnet/core/blob/master/release-notes/preview4-download.md
- Open a PS prompt, navigate to SlackAPI root folder and launch ./build.ps1

To summarize:
- SlackApi.NetCore.sln, SlackApi.NetCore.csproj, SlackApi.Tests.NetCore.csproj are new files compatible with VS2017. If we have to add a reference, we have to do it in both VS2015 and VS2017 projects. If we add a file in VS2015 solution, it's automatically added in VS2017 (because new csproj supports wildcards)
- Some files are updated to support all Frameworks (mainly reflection related stuff)
- build.cake is a new file used to compile, test and pack SlackAPI
- build.ps1 is a boostrapper from CakeBuild project
- appveyor.yml is updated to install correct .NetCore version (not available on AppVeyor right now), use Cake script and publish artifacts on myGet. We can update the yml to publish artifacts on NuGet too

Note:
- I included a fix similar to #89 in the changes

### CODE REVIEW

**Comment by ibruynin** [comment]
Hi @gpailler 

when opening the VS2017 solution I get the following errors:

\\vmware-host\Shared Folders\Documents\Git\GitHub\SlackAPI\SlackAPI\SlackAPI.NetCore.csproj : error  : The project file cannot be found.
\\vmware-host\Shared Folders\Documents\Git\GitHub\SlackAPI\SlackAPI.Tests\SlackAPI.Tests.NetCore.csproj : error  : The project file cannot be found.

I'll try to resolve these myself, so this is FYI.

**Comment by gpailler** [comment]
@ibruynin thanks for the comment. I fixed the paths

**Comment by ibruynin** [comment]
Basic bot functionalities now work on our raspberry Pi with win 10 iot core and .netstandard1.3
nice work @gpailler

**Comment by gpailler** [comment]
@ibruynin Nice!. Thanks for the feedback

File: SlackAPI.Tests/JoinDirectMessageChannel.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public void ShouldJoinDirectMessageChannel()
             JoinDirectMessageChannelResponse actual = null;
 
             string userName = this.fixture.Config.DirectMessageUser;
-            string user = client.Users.First(x => x.name.Equals(userName, StringComparison.InvariantCultureIgnoreCase)).id;
+            string user = client.Users.First(x => x.name.Equals(userName, StringComparison.OrdinalIgnoreCase)).id;
```
Comment (Inumedia) [inline SlackAPI.Tests/JoinDirectMessageChannel.cs]: Why was this changed to Ordinal instead of Invariant?

Comment (gpailler) [inline SlackAPI.Tests/JoinDirectMessageChannel.cs]: Invariant is not supported in .NetCore. And according to this [issue](https://github.com/dotnet/corefx/issues/9801), InvariantCultureIgnoreCase usage is often to what we want

**Comment by gpailler** [comment]
I fixed small conflict related to latest PR applied on master

**Comment by gpailler** [comment]
@Inumedia Probably you should wait before merging this PR. VS2017 final will be available today and I will update appveyor script as soon as they publish official image for VS2017 (it will simplify the appveyor.yml script)

**Comment by Inumedia** [comment]
:+1:

**Comment by gpailler** [comment]
@Inumedia PR updated with full VS2017 support
I think you can merge this PR as is or if you have already switched to VS2017, I can replace VS2015 compatible sln/csproj with new ones.

**Comment by andymac4182** [comment]
Is this far away from being pulled in?
